src/bundle/Controller/AssetController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUiBundle\Controller;
+
+use eZ\Publish\Core\FieldType\Image\Value as ImageValue;
+use eZ\Publish\Core\FieldType\ImageAsset\Mapper as ImageAssetMapper;
+use EzSystems\EzPlatformAdminUi\Form\Data\Asset\AssetUploadData;
+use EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class AssetController extends Controller
+{
+    /** @var \EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory */
+    private $formFactory;
+
+    /** @var \eZ\Publish\Core\FieldType\ImageAsset\Mapper */
+    private $imageAssetMapper;
+
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\Form\Factory\FormFactory $formFactory
+     * @param \eZ\Publish\Core\FieldType\ImageAsset\Mapper $assetMapper
+     */
+    public function __construct(FormFactory $formFactory, ImageAssetMapper $assetMapper)
+    {
+        $this->formFactory = $formFactory;
+        $this->imageAssetMapper = $assetMapper;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentType
+     */
+    public function uploadImageAction(Request $request): Response
+    {
+        $data = new AssetUploadData();
+
+        $form = $this->formFactory->assetUpload($data, 'form');
+        $form->handleRequest($request);
+        if ($form->isValid()) {
```

Comment:
I believe it's advised by Symfony to use `$form->isSubmitted() && $form->isValid()` together.